osh/hub/osh_xmlrpc/worker.py

```diff
@@ -101,6 +103,11 @@ def finish_task(request, task_id):
     exclude_dirs = AppSettings.settings_get_results_tb_exclude_dirs()
     td = TaskResultsProcessor(task, base_task, exclude_dirs)
     td.unpack_results()
+
+    msg = message.Message(topic=u'task.finish', headers={u'task_id': task_id}, body={})
```

Comment:
This line should be probably moved in the below if condition. And we should add reference to `scan-results.js` for full scan and `added.js`  for a differential scan.

Reply:
@lachmanfrantisek @lbarcziova

Once you have the results in json format, follow the steps mentioned in https://github.com/packit/dashboard/issues/441#issuecomment-2301884237

Reply:
Just to check. The `task_id` is the ID of the OpenScanHub, right?

Reply:
The id is the crucial one, but here are a few other things I think might be useful to have in the body (if possible):
* Link to the task we can provide to the user to check details.
  * Do you provide a standardised output format (?SARIF) we can link as well?
* Info about the scan input (e.g. package NVR, architecture, SRPM link)

Reply:
Agreed that the task ID with the state change are the most important ones, and then some info we can work with regarding the outputs. Not sure if the input info is needed really, but we can discuss that in person.

Reply:
> Agreed that the task ID with the state change are the most important ones, and then some info we can work with regarding the outputs. Not sure if the input info is needed really, but we can discuss that in person.

If a differential scan is successful, you would get a reference to added and fixed defects in the body. Currently it looks like:

```
Body: {                                                                                                                                                                                                                                     
    "added.js": "http://openscanhub.stg.fedoraproject.org/task/1864/log/added.js?format=raw",
    "fixed.js": "http://openscanhub.stg.fedoraproject.org/task/1864/log/fixed.js?format=raw",                                                                                                                                               
    "scan-results.js": "http://openscanhub.stg.fedoraproject.org/task/1864/log/units-2.22-6.fc39/scan-results.js?format=raw"
}  
```

You may want to upload both added and fixed findings to CodeQL.